astropy/units/tests/test_format.py

```diff
@@ -419,27 +419,41 @@ def test_scaled_dimensionless():
 
 
 def test_deprecated_did_you_mean_units():
-    with pytest.raises(ValueError) as exc_info:
+    with pytest.raises(ValueError,
+                       match=r'Crab \(deprecated\) or mCrab \(deprecated\)'):
+        u.Unit('crab', format='ogip')
+
+    with pytest.raises(ValueError, match=r'Did you mean Angstrom or angstrom'):
         u.Unit('ANGSTROM', format='fits')
-    assert 'Did you mean Angstrom or angstrom?' in str(exc_info.value)
 
-    with pytest.raises(ValueError) as exc_info:
-        u.Unit('crab', format='ogip')
-    assert 'Crab (deprecated)' in str(exc_info.value)
-    assert 'mCrab (deprecated)' in str(exc_info.value)
+    with pytest.raises(ValueError,
+                       match=r'not supported by the CDS SAC standard') as e:
+        u.Unit('ANGSTROM', format='cds')
+    assert str(e.value).count('Angstrom') == 5
 
-    with catch_warnings() as w:
+    with pytest.raises(ValueError,
+                       match=r'not supported by the VOUnit standard') as e:
         u.Unit('ANGSTROM', format='vounit')
+    assert str(e.value).count('nm') == 1
 
+    with catch_warnings() as w:
```

Comment:
There is also a `pytest.warns`, which works just like `pytest.raises`

Reply:
Thanks for reminding me I had actually been a quite vocal proponent of using `pytest.warns` wherever a warning is really expected and required! ;-)
This time I had just lazily copied the entire template from another test...